opal/datatype/opal_copy_functions_heterogeneous.c

```diff
@@ -66,6 +67,36 @@ static inline void opal_dt_swap_bytes(void *to_p, const void *from_p, const size
     }
 }
 
+static inline void opal_dt_swap_bytes_inplace(void *buf_p, const size_t size,
```

Comment:
Thanks, I like the swap you linked to.  For some of the cases it's way faster.  My performance tests are all over the map with it though.   On my mac:
4b: 62.7 --> 5.07
8b: 110.7 --> 9.89
16b: 21.18 --> 25.5
Then on a ppc machine:
4b: 37.58 --> 59.5
8b: 106.4 --> 59.7
16b: 294.4 --> 80.0
So sometimes 4x-10x speedup, and occasionally maybe 50% slower.

The reason I added an in-place byte swap wasn't for performance though, it was just because the macro currently has to do the f80/f128 conversion and the byte swapping as two separate steps, and we can't modify the 'from' buffer, so depending on the order we need for the operations, I needed a byte-swap that could run in place on the 'to' buffer

Also on my ppc machine the __float128 conversion was a bit slower than the manual approach (I don't have that number in front of me, but maybe 20% slower).  I'm still inclined to use it just because it's guaranteed to be correct.

But overall I'm leaning toward not trying to address performance too much in this commit.